sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/redis/passwordless/jedis/AzureJedisConnectionFactory.java

```diff
@@ -0,0 +1,325 @@
+    }
+
+    @Override
+    public void destroy() {
+        if (getUsePool() && pool != null) {
+
+            try {
+                pool.destroy();
+            } catch (Exception ex) {
+                LOGGER.warn("Cannot properly close Jedis pool", ex);
+            }
+            pool = null;
+        }
+
+        this.destroyed = true;
+    }
+
+    @Override
+    public DataAccessException translateExceptionIfPossible(RuntimeException ex) {
+        throw new UnsupportedOperationException("This operation is not supported");
+    }
+
+    @Override
+    public RedisConnection getConnection() {
+        assertInitialized();
+
+        Jedis jedis = fetchJedisConnector();
+
+        JedisClientConfig sentinelConfig = this.jedisClientConfig;
+
+        JedisConnection connection = (getUsePool() ? new AzureJedisConnection(jedis, pool, this.jedisClientConfig, sentinelConfig)
+            : new AzureJedisConnection(jedis, null, this.jedisClientConfig, sentinelConfig));
+        connection.setConvertPipelineAndTxResults(convertPipelineAndTxResults);
+        return postProcessConnection(connection);
+    }
+
+    /**
+     * Specifies if pipelined results should be converted to the expected data type. If false, results of
+     * {@link JedisConnection#closePipeline()} and {@link JedisConnection#exec()} will be of the type returned by the
+     * Jedis driver.
+     *
+     * @param convertPipelineAndTxResults Whether or not to convert pipeline and tx results.
+     */
+    public void setConvertPipelineAndTxResults(boolean convertPipelineAndTxResults) {
+        this.convertPipelineAndTxResults = convertPipelineAndTxResults;
+    }
+
+    /**
+     * Indicates the use of a connection pool.
+     * <p>
+     * Applies only to single node Redis.
+     *
+     * @return the use of connection pooling.
+     */
+    public boolean getUsePool() {
+        return clientConfiguration.isUsePooling();
+    }
+
+    /**
+     * Returns the index of the database.
+     *
+     * @return the database index.
+     */
+    public int getDatabase() {
+        return RedisConfiguration.getDatabaseOrElse(standaloneConfig, standaloneConfig::getDatabase);
+    }
+
+    /**
+     * Returns the port used to connect to the Redis instance.
+     *
+     * @return the Redis port.
+     */
+    public int getPort() {
+        return standaloneConfig.getPort();
+    }
+
+    /**
+     * @return {@literal true} to use SSL, {@literal false} to use unencrypted connections.
+     */
+    public boolean isUseSsl() {
+        return clientConfiguration.isUseSsl();
+    }
+
+    /**
+     * Returns the Redis hostname.
+     *
+     * @return the hostName.
+     */
+    public String getHostName() {
+        return standaloneConfig.getHostName();
+    }
+
+    /**
+     * Returns the password used for authenticating with the Redis server.
+     *
+     * @return password for authentication.
+     */
+    @Nullable
+    public String getPassword() {
```

Comment:
Do you think we can skip all these getter methods in this first version?

Reply:
I think we can make it package level?

Reply:
I am trying to align the AzureJedisConnectionFactory with JedisConnectionFactory, so expose these getter methods.

Reply:
I understand that both `JedisConnectionFactory` and `LettuceConnectionFactory` are implementations of `RedisConnectionFactory`. But the `RedisConnectionFactory` doesn't have these APIs, so a user who develops against the `RedisConnectionFactory` won't know these getters. I think we can skip these getters in the first release, and add them later when there is an ask.

Reply:
ok, I will refactor this.